velox/type/Type.cpp

```diff
@@ -40,6 +41,63 @@ bool isColumnNameRequiringEscaping(const std::string& name) {
 
 namespace facebook::velox {
 
+TypePtr Type::resolveUnknownTypes(const TypePtr& a, const TypePtr& b) {
+  if (a->isUnKnown()) {
+    return b;
+  }
+
+  if (b->isUnKnown()) {
+    return a;
+  }
+
+  if (a->kind() != b->kind()) {
+    return nullptr;
+  }
+
+  if (a->isPrimitiveType()) {
+    return a;
```

Comment:
We probably need to check for `DECIMAL` and make sure the precision and scale are matching.